lib/types.ts

```diff
@@ -271,23 +273,74 @@ export type DeepOmit<T extends DeepOmitModify<Filter>, Filter> = T extends Built
         {
           [K in Extract<RequiredKeys<T>, keyof Filter>]: Filter[K] extends true
             ? never
-            : T[K] extends DeepOmitModify<Filter[K]>
+            : T[K] extends DeepModify<Filter[K]>
             ? DeepOmit<T[K], Filter[K]>
             : T[K];
         },
         never
       >;
-type DeepOmitModify<T> =
+
+/** Recursively pick deep properties */
+export declare type DeepPick<T, Filter extends DeepModify<T>> = T extends Builtin
+  ? T
+  : T extends Map<infer KeyType, infer ValueType>
+  ? ValueType extends DeepModify<Filter>
+    ? Map<KeyType, DeepPick<ValueType, Filter>>
+    : T
+  : T extends ReadonlyMap<infer KeyType, infer ValueType>
+  ? ValueType extends DeepModify<Filter>
+    ? ReadonlyMap<KeyType, DeepPick<ValueType, Filter>>
+    : T
+  : T extends WeakMap<infer KeyType, infer ValueType>
+  ? ValueType extends DeepModify<Filter>
+    ? WeakMap<KeyType, DeepPick<ValueType, Filter>>
+    : T
+  : T extends Set<infer ItemType>
+  ? ItemType extends DeepModify<Filter>
+    ? Set<DeepPick<ItemType, Filter>>
+    : T
+  : T extends ReadonlySet<infer ItemType>
+  ? ItemType extends DeepModify<Filter>
+    ? ReadonlySet<DeepPick<ItemType, Filter>>
+    : T
+  : T extends WeakSet<infer ItemType>
+  ? ItemType extends DeepModify<Filter>
+    ? WeakSet<DeepPick<ItemType, Filter>>
+    : T
+  : T extends Array<infer ItemType>
+  ? ItemType extends DeepModify<Filter>
+    ? Array<DeepPick<ItemType, Filter>>
+    : T
+  : T extends Promise<infer ItemType>
+  ? ItemType extends DeepModify<Filter>
+    ? Promise<DeepPick<ItemType, Filter>>
+    : T
+  : Filter extends Record<string, unknown>
+  ? {
+      [K in keyof Filter & OptionalKeys<T>]?: Filter[K] extends true ? T[K] : DeepPick<T[K], Filter[K]>;
```

Comment:
TODO:

1. Add test cases with `never`
2. Add `IsNever` check after merge of https://github.com/krzkaczor/ts-essentials/pull/251

Reply:
- [x] Added test cases for `never` filter
- [x] Don't need to unify check for the filter as it's already mentioned once in `DeepModify`